### DISCUSSION TITLE
Signature declined in web3modal's siwe

### DISCUSSION THREAD

**Discussion by the-content** [WalletConnect/walletconnect-monorepo#4299] [Developer Support]
package version：
"@web3modal/siwe": "^4.0.13",
"@web3modal/wagmi": "^3.5.0",
"viem": "^1.20.1",
"wagmi": "^1.4.12"
"next": "14.0.4"

When I follow the official siwe integration, when I click on sign, I can get the nonce through next-auth's getCsrfToken method, but immediately after that I get the ``Signature declined`` error
With this error I searched for the error and printed the full error:
<img width="656" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/65217158/b41a7604-9f08-409e-940e-1c0478db4ce5">
Complete Error:
<img width="730" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/65217158/3dd22abb-230d-422b-b7cf-4257e74f2af0">

I found the exact cause of the error step by step by searching globally for this text, and there was a problem with the signIn method in the client.js file of the npm package:

```
async signIn() {
        const nonce = await this.methods.getNonce();
        const { address } = AccountController.state;
        if (!address) {
            throw new Error('An address is required to create a SIWE message.');
        }
        const chainId = HelpersUtil.caipNetworkIdToNumber(NetworkController.state.caipNetwork?.id);
        if (!chainId) {
            throw new Error('A chainId is required to create a SIWE message.');
        }
        const message = this.methods.createMessage({ address, nonce, chainId });
        const signature = await ConnectionController.signMessage(message);
        const isValid = await this.methods.verifyMessage({ message, signature });
        if (!isValid) {
            throw new Error('Error verifying SIWE signature');
        }
        const session = await this.methods.getSession();
        if (!session) {
            throw new Error('Error verifying SIWE signature');
        }
        if (this.methods.onSignIn) {
            this.methods.onSignIn(session);
        }
        RouterUtil.navigateAfterNetworkSwitch();
        return session;
    }
```

```
const { address } = AccountController.state;
```
Where print address get get is empty, so it reports an error ```An address is required to create a SIWE message.```

Why is it failing to get the address of the currently connected account?
But my front end already reads and displays the address of the connection, what's wrong?

<img width="729" alt="image" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/65217158/28e21fb8-2264-452d-a7ed-264572cc721b">

**Comment by glitch-txs** [comment]
Hi, could you share a minimal reproducible example and steps to reproduce the issue? Thanks!
